### ISSUE TITLE
Revise Regex usage in TrimStackTrace

### ISSUE THREAD

**Issue by Youssef1313** [microsoft/testfx#4266]
https://github.com/microsoft/testfx/blob/beb890d4fe699cf718748884c1e642e28239c1fc/src/Adapter/MSTest.TestAdapter/Execution/ExceptionHelper.cs#L109

It doesn't make much sense to me to use Regex here.

Not so much allocations but would be good to reduce:

![Image](https://github.com/user-attachments/assets/2597b08c-7116-48c5-979d-77dbb1bb6e0e)

**Comment by Youssef1313** [comment]
This can even be more efficient than `string.Split`.

We can do the loop ourselves and use spans to not allocate anything here.

**Comment by nohwnd** [comment]
Agree. Probably was used because it has overload that takes string, rather than having to wrap the string into array to call string.split.
